elobot.py

```diff
@@ -25,23 +25,44 @@ def __init__(self, slack_client, channel, config):
         self.slack_client = slack_client
         self.channel = channel
         self.config = config
-        self.slack_client.rtm_connect()
+        self.connect()
         self.run()
-        
+
+    def connect(self):
+        sleeptime = 0.1
+
+        while True:
+            try:
+                self.slack_client.rtm_connect()
+                break
+            except:
+                time.sleep(sleeptime)
+
+                #Exponential back off with a max wait of 30s
+                if sleeptime < 30:
```

Comment:
Could do it either way. It doesn't matter to me if it sits there and retries forever. Eventually we'd probably want it to go back online.